src/kinetics/LinearBurkeRate.cpp

```diff
@@ -0,0 +1,354 @@
+    ReactionRate::setParameters(node, rate_units);
+    if (!node.hasKey("collider-list")) {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "Incorrect YAML input for LMR-R reaction. Please review implementation guide.");
+    }
+    auto colliders = node["collider-list"].asVector<AnyMap>();
+    if (!colliders[0].hasKey("collider")) {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "Incorrect YAML input for LMR-R reaction. Please review implementation guide.");
+    }
+    else if (colliders[0]["collider"].as<string>() != "M") {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "The first species defined in LMR-R YAML input must be 'M'. Please review implementation guide.");
+    }
+    else if (!colliders[0].hasKey("eig0") && !colliders[0].hasKey("eps")) {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "A third-body efficiency (eps) or ME eigenvalue (eig0) must be provided for collider M. Please review implementation guide.");
+    }
+    else if (colliders[0].hasKey("eps")) {
+        if (colliders[0]["eps"]["A"] != 1 || colliders[0]["eps"]["b"] != 0 || colliders[0]["eps"]["Ea"] != 0) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "The third-body efficiency (eps) must be entered for M as 'eps: {A: 1, b: 0, Ea: 0}'. Please review implementation guide.");
+        }
+    }
+    if (colliders[0].hasKey("rate-constants")) {
+        rateObj_M = PlogRate(colliders[0], rate_units);
+        dataObj_M = PlogData();
+    }
+    else if (colliders[0].hasKey("Troe")) {
+        // Value of "type" is unimportant; just needed to make falloff.cpp run
+        colliders[0]["type"];
+        rateObj_M = TroeRate(colliders[0], rate_units);
+        dataObj_M = FalloffData();
+    }
+    else if (colliders[0].hasKey("pressure-range")) {
+        rateObj_M = ChebyshevRate(colliders[0], rate_units);
+        dataObj_M = ChebyshevData();
+    }
+    else {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "The M-collider must be specified in a pressure-dependent-Arrhenius (PLOG), falloff (Troe form), or Chebyshev format.");
+    }
+    string eig_eps_key;
+    if (colliders[0].hasKey("eig0") && !colliders[0].hasKey("eps")) {
+        eig_eps_key = "eig0";
+    }
+    else if (colliders[0].hasKey("eps") && !colliders[0].hasKey("eig0")) {
+        eig_eps_key = "eps";
+    }
+    else{
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "Cannot have both eig0 and eps chosen for M. Any additional colliders must make same choice as that of M. Please review implementation guide.");
+    }
+    AnyMap params;
+    params["A"] = 1.0;
+    params["b"] = 0.0;
+    params["Ea"] = 0.0;
+    epsObj_M = ArrheniusRate(AnyValue(params), colliders[0].units(), eps_units);
```

Comment:
Since you're always converting to the form that uses epsilon and $\epsilon_M=1$ by definition, you could just eliminate the `epsObj_M` object.

Reply:
It would be much more convenient to keep epsObj_M as an ArrheniusRate object, as changing it would complicate the structure of evalFromStruct()

Reply:
I have found a middle-ground solution, where the use of epsObj_M in setParameters() for the colliders containing eig or eps but laking k(TP) is maintained, but the separate calculation of eps_M in evalFromStruct() for the unspecified "pure-M" colliders has been eliminated. The changes can be seen in commit #9e71a5d3